src/infer_trace.h

```diff
@@ -69,26 +70,34 @@ class InferenceTrace {
   void SetModelVersion(int64_t v) { model_version_ = v; }
   void SetRequestId(const std::string& request_id) { request_id_ = request_id; }
   void SetContext(const std::string& context) { context_ = context; }
+  void RecordActivityName(uint64_t timestamp_ns, std::string activity_name);
 
   // Report trace activity.
   void Report(
-      const TRITONSERVER_InferenceTraceActivity activity, uint64_t timestamp_ns)
+      const TRITONSERVER_InferenceTraceActivity activity, uint64_t timestamp_ns,
+      std::string activity_name = "")
   {
     if ((level_ & TRITONSERVER_TRACE_LEVEL_TIMESTAMPS) > 0) {
+      if (activity == TRITONSERVER_TRACE_CUSTOM_ACTIVITY &&
+          !activity_name.empty()) {
```

Comment:
Should custom activity always be added to context even the name is empty? Otherwise `activity_fn` will record an custom activity that can't be found in context

Reply:
implemented suggested version